compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -353,26 +450,179 @@ LogicalResult getUserTiles(
+
+  for (AMDAIE::CoreOp coreOp : coreOps) {
+    DenseMap<Value, std::tuple<AMDAIE::LogicalObjectFifoFromMemrefOp,
+                               AMDAIE::MemoryAccess>>
+        memrefToLogicalObjectFifo;
+    // First walk to collect consume/produce DMA accesses and map respective
+    // memrefs to logical objectifos.
+    coreOp->walk([&](Operation *op) {
+      // TODO(jornt): can we avoid produce/consume?
+      if (auto consumeOp = dyn_cast<AMDAIE::LogicalObjectFifoConsume>(op)) {
+        Value targetMemref =
+            consumeOp.getDmaCpyNdOp().getTargetObjectFifo().getMemref();
+        memrefToLogicalObjectFifo[targetMemref] =
+            std::make_pair(consumeOp.getDmaCpyNdOp().getTargetObjectFifo(),
+                           AMDAIE::MemoryAccess::Read);
+      } else if (auto produceOp =
+                     dyn_cast<AMDAIE::LogicalObjectFifoProduce>(op)) {
+        Value sourceMemref =
+            produceOp.getDmaCpyNdOp().getSourceObjectFifo().getMemref();
+        memrefToLogicalObjectFifo[sourceMemref] =
+            std::make_pair(produceOp.getDmaCpyNdOp().getSourceObjectFifo(),
+                           AMDAIE::MemoryAccess::Write);
+      }
+    });
+
+    WalkResult res = coreOp->walk([&](Operation *op) {
+      if (auto linalgOp = dyn_cast<linalg::LinalgOp>(op)) {
+        for (auto &&[idx, operand] :
+             llvm::enumerate(linalgOp->getOpOperands())) {
+          if (memrefToLogicalObjectFifo.contains(operand.get())) {
+            rewriter.setInsertionPointToStart(coreOp.getBody());
+            std::tuple<AMDAIE::LogicalObjectFifoFromMemrefOp,
+                       AMDAIE::MemoryAccess>
+                value = memrefToLogicalObjectFifo[operand.get()];
+            rewriter.create<AMDAIE::LogicalObjectFifoAccessOp>(
+                rewriter.getUnknownLoc(), std::get<0>(value),
+                std::get<1>(value));
+            // TODO(jornt): Temporary, enable after access operations are used
+            // for inserting synchronization stubs instead of consume/produce.
+            // linalgOp->setOperand(idx, accessOp);
+          } else if (auto type =
+                         llvm::dyn_cast<MemRefType>(operand.get().getType())) {
+            Value memref = operand.get();
+            rewriter.setInsertionPoint(coreOp);
+            auto logicalObjectFifo =
+                rewriter.create<AMDAIE::LogicalObjectFifoFromMemrefOp>(
+                    rewriter.getUnknownLoc(), LogicalObjectFifoType::get(type),
+                    memref);
+            rewriter.setInsertionPointToStart(coreOp.getBody());
+            rewriter.create<AMDAIE::LogicalObjectFifoAccessOp>(
+                rewriter.getUnknownLoc(), logicalObjectFifo,
+                AMDAIE::MemoryAccess::None);
+            // TODO(jornt): Temporary, enable after access operations are used
+            // for inserting synchronization stubs instead of consume/produce.
+            // linalgOp->setOperand(idx, accessOp);
```

Comment:
Temporary to avoid `LogicalObjectFifoAccessOp` uses for now. Will be uncommented in a follow-up, see https://github.com/nod-ai/iree-amd-aie/compare/jornt_cpp_pipeline?expand=1